File: 10th/hyeongsh/longest_even_continuous_partial_sequence_22862.cpp

#include <iostream>

int	arr[1000001];

int	main(void)
{
	int	s, k;
	std::cin >> s >> k;
	for (int i = 0; i < s; i++)
		std::cin >> arr[i];
	int	*st = arr;
	int	*en = arr;
	int	del = 0;
	int	max = -1;
	while (*st != 0)
	{
		while (*en != 0)
		{
			if (*en % 2 == 1 && del < k)
				del++;
			else if (*en % 2 == 1)
				break ;
			en++;
		}
		//std::cout << "st: " << st - arr << "  en: " << en - arr << std::endl;
		if (max < en - st - del)
			max = en - st - del;
		if (*st % 2 == 1)
			del--;
		st++;
	}
	std::cout << max << std::endl;
}

File: 10th/jongmlee/LongestEvenContinuousPartialSequence22862.java
import java.io.*;
import java.util.*;

public class LongestEvenContinuousPartialSequence22862 {
	public static void main(String args[]) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		int n = Integer.parseInt(st.nextToken());
		int k = Integer.parseInt(st.nextToken());
		int[] arr = new int[n];
		st = new StringTokenizer(br.readLine());
		for (int i = 0; i < n; i++) {
			arr[i] = Integer.parseInt(st.nextToken());
		}
		int p1 = 0;
		int p2 = 0;
		int maxLen = -1;
		int OddCnt = 0;
		for (; p2 < n; p2++) {
			if (arr[p2] % 2 == 1) {
				OddCnt++;
			}
			if (OddCnt > k) {
				maxLen = Math.max((p2 - p1) - (OddCnt - 1), maxLen);
				for (; arr[p1] % 2 == 0; p1++);
				--OddCnt;
				++p1;
			}
		}
		maxLen = Math.max(p2 - p1 - OddCnt, maxLen);
		System.out.println(maxLen);
	}
}
